exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -186,27 +186,57 @@ func (ampmr *AzureMachinePoolMachineController) Reconcile(ctx context.Context, r
 		return ctrl.Result{}, nil
 	}
 
-	azureClusterName := client.ObjectKey{
-		Namespace: machine.Namespace,
-		Name:      cluster.Spec.InfrastructureRef.Name,
-	}
+	var clusterScope azure.ClusterScoper
+	switch cluster.Spec.InfrastructureRef.Kind {
+	case "AzureCluster":
+		azureClusterName := client.ObjectKey{
+			Namespace: machine.Namespace,
+			Name:      cluster.Spec.InfrastructureRef.Name,
+		}
 
-	azureCluster := &infrav1.AzureCluster{}
-	if err := ampmr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
-		logger.Info("AzureCluster is not available yet")
-		return reconcile.Result{}, nil
-	}
+		azureCluster := &infrav1.AzureCluster{}
+		if err := ampmr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
+			logger.Info("AzureCluster is not available yet")
+			return reconcile.Result{}, nil
+		}
 
-	logger = logger.WithValues("AzureCluster", azureCluster.Name)
+		logger = logger.WithValues("AzureCluster", azureCluster.Name)
 
-	// Create the cluster scope
-	clusterScope, err := scope.NewClusterScope(ctx, scope.ClusterScopeParams{
-		Client:       ampmr.Client,
-		Cluster:      cluster,
-		AzureCluster: azureCluster,
-	})
-	if err != nil {
-		return reconcile.Result{}, err
+		// Create the cluster scope
+		scoper, err := scope.NewClusterScope(ctx, scope.ClusterScopeParams{
+			Client:       ampmr.Client,
+			Cluster:      cluster,
+			AzureCluster: azureCluster,
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		clusterScope = scoper
+	case "AzureManagedCluster":
+		azureManagedControlPlaneName := client.ObjectKey{
+			Namespace: machine.Namespace,
+			Name:      cluster.Spec.ControlPlaneRef.Name,
+		}
+		azureManagedControlPlane := &infrav1.AzureManagedControlPlane{}
+		if err := ampmr.Client.Get(ctx, azureManagedControlPlaneName, azureManagedControlPlane); err != nil {
+			logger.Info("AzureManagedControlPlane is not available yet")
+			return reconcile.Result{}, nil
```

Comment:
ditto above re: watching AzureManagedControlPlane

Reply:
This controller currently doesn't watch AzureClusters either, since it watches AzureMachinePools that should be enough though, I think?

Reply:
The helper function now returns an error for this now so it should requeue right away (in practice I don't expect this error to happen outside of the initial yaml apply, unless the user forgot something in which case we want to alert them)